### PR TITLE
fix(#4098): a declared control inbox is exclusive — a missing secret refuses instead of self-delivering

### PR DESCRIPTION
**Pull request by rbuergi**
Part of #4098 (P3b). Fixes the defect filed in [the 2026-09-15 evidence comment](https://github.com/Systemorph/MeshWeaver/issues/4098) — a control plane silently talking to itself, sitting directly across this issue's own acceptance step.

## The defect

`SelfUpdateHandover.RouteFor` fell through to `Route.Local` when a control inbox was declared but `Hosting:ControlInbox:Secret` was absent:

```csharp
if (settings.Url is { Length: > 0 } && settings.SecretPresent) return Route.Post;
if (settings.LocalTargetListed && settings.LocalSecretKey is { Length: > 0 } && settings.LocalSecretPresent) return Route.Local;
return Route.None;
```

The method's own doc comment already stated the intended rule — *"an instance that declares a control inbox is a consumer, **whatever it also lists**"* — but the code only honoured it when the secret was present.

`Deployments/build` is exactly that shape: `Hosting__ReportTo` derives the inbox URL ✓, `Hosting__ControlInbox__Secret` is **absent** ✗, and `build` legitimately **lists** `Hosting/PlatformBuilds` with a declared `SecretConfigKey` whose secret **is** mounted — it owns the fleet's build queue. So the moment this issue's own acceptance step (chart pin + Reconcile → `SelfUpdate__CanPatch=false`) runs there:

- `ApplyModeFor` → `ControlLane`
- `DeliverLocally` stores the release in **build's own** inbox
- build's own watcher (`PlatformBuildInboxWatcher.PlanFor`) classifies `self-update-available` as a non-build event and **deletes** it
- the boot line reads `apply=control-lane (a detected release is handed to this instance's own Hosting/PlatformBuilds inbox; the control plane rolls)`, `Announce` records `Outcome(Local, …, "signature verified")`, and `Missing()` returns null so **nothing names the absent key**

A hand-over that reaches nobody, reported as a verified success — and #4098's acceptance evidence would have been read off exactly that line. A verification step that cannot fail is not one.

## The fix

A declared control inbox is now **exclusive**. The three routes, as now decided:

| shape | route | what the operator reads |
|---|---|---|
| URL declared/derived **+** secret present | `Post` | `apply=control-lane (… handed to https://memex.systemorph.com/api/hooks/Hosting/PlatformBuilds …)` |
| URL declared/derived, secret **absent** | `None` → `DetectOnly` | `apply=detect-only (no control inbox: Hosting:ReportTo names the control instance but Hosting:ControlInbox:Secret is empty, so nothing could be signed)` |
| **no** URL + listed target declaring a key whose secret is present | `Local` | unchanged — the control instance's route; it names no control instance above it |

`Missing()` already ordered the URL check ahead of the local ones, so the refusal sentence needed no new text — it just became reachable. `Missing()` now also takes the **same** "a URL is declared" test the route does (`is { Length: > 0 }` rather than `is not null`), so an instance that declared neither URL key is never blamed for them.

The config-only alternative — mount `Hosting__ControlInbox__Secret` on `build`'s record — fixes one record and leaves the silent self-delivery reachable for the next instance that owns an inbox. That mapping is still owed (it is a tracked item on #4098, in `Systemorph/Memex`); this makes the absence a refusal rather than a fiction.

## Tests — written failing first

Both new cases returned `Local` before the fix:

```
Expected value to be None because an instance that names a control instance is a CONSUMER — its own
inbox is not a control plane, and the event it stores there is deleted by its own watcher, but found Local.
```

- `ADeclaredControlInbox_IsExclusive_NeverFallingThroughToLocalDelivery` — the three shapes above, pure.
- `AnInstanceThatNamesAControlInstance_AndOwnsAnInbox_RefusesRatherThanTalkingToItself` — the measured `build` configuration driven through `ReadSettings`, asserting `Route.None`, `SelfUpdateApply.DetectOnly`, and a missing sentence that names `Hosting:ControlInbox:Secret` + `Hosting:ReportTo` and **not** `Hosting/PlatformBuilds` (the target it drains for the fleet is not what stands between it and the control lane).

The existing tests could not see this: `SelfUpdateHandoverTest` built the local case with `Url == null`, and the URL-without-secret case only with the local target **unlisted**.

## Verification

- `dotnet build memex/Memex.Portal.Shared/Memex.Portal.Shared.csproj -c Release -warnaserror` → `0 Error(s)`, `0 Warning(s)`
- `dotnet build test/Memex.Portal.Shared.Test/…csproj -c Release -warnaserror` → `0 Error(s)`
- `dotnet build test/MeshWeaver.Documentation.Test/…csproj -c Release -warnaserror` → `0 Error(s)`
- `Memex.Portal.Shared.Test` (whole project, rebuilt after the doc asset changed): **1574 passed, 0 failed**
- `MeshWeaver.Documentation.Test` (whole project): **469 passed, 0 failed** (link integrity + What's New integrity)

No public type or member removed and no interface member added, so neither the cross-repo pair gate nor the implementers gate applies — the behaviour change is behind an unchanged signature.

## Work products conserved

- `Doc/Architecture/SelfUpdateControlLane` — the Routes section now states the exclusivity rule, the `build` shape that made it load-bearing, and why the refusal lives in the route rather than in one record's vault mapping.
- What's New (`Category: Fix`) — what an operator of such a portal now reads instead of a false hand-over.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/MeshWeaver.Documentation/Data/WhatsNew/2026-09-15-a-portal-that-cannot-hand-over-an-update-says-so-instead-of-delivering-to-itself.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+Name: A portal that cannot hand over an update says so instead of delivering to itself
+Category: Fix
+Description: A portal that names a control instance but has no signing secret now reports detect-only and names the missing key, instead of quietly storing the release in its own inbox and reporting a successful hand-over.
```
[the remark this answers is not in the document]
Comment (rbuergi) [reply src/MeshWeaver.Documentation/Data/WhatsNew/2026-09-15-a-portal-that-cannot-hand-over-an-update-says-so-instead-of-delivering-to-itself.md]: Valid — fixed in baf2c94c5e.

You are right that `ApplyModeFor` deliberately keeps `SelfPatch` when the chart (`SelfUpdate:CanPatch`) and the updater both allow it, so a portal on a pre-#4291 chart is untouched by this. The entry now opens by scoping itself ("Portals are moving off applying platform updates themselves. **Where that has happened** …"), states the unaffected case on its own line ("*A portal that still applies its own updates is unaffected by everything below*"), and the `Description` carries the same scope so the gallery card cannot be read as "every portal now hands over".

File: test/Memex.Portal.Shared.Test/SelfUpdateHandoverTest.cs [hunk 2]
```diff
@@ -72,6 +72,76 @@ public void TheControlInstanceDeliversLocally_OnlyWithItsTargetListedDeclaringAK
...
+    {
+        var config = new ConfigurationBuilder().AddInMemoryCollection(new Dictionary<string, string?>
+        {
+            [SelfUpdateHandover.DeploymentKey] = Deployment,
+            [SelfUpdateHandover.ReportToKey] = "https://memex.systemorph.com",
+            // No Hosting:ControlInbox:Secret — the record does not map it.
+            ["WebhookInbox:Targets:0"] = SelfUpdateHandover.InboxTarget,
+            ["WebhookInbox:Targets:0:SecretConfigKey"] = SelfUpdateHandover.LocalSecretKey,
+            [SelfUpdateHandover.LocalSecretKey] = "fleet-secret",
+        }).Build();
```
[the remark this answers is not in the document]
Comment (rbuergi) [reply test/Memex.Portal.Shared.Test/SelfUpdateHandoverTest.cs:123]: Not taking this one — the premise does not hold and the change would make the new test the odd one out.

- **"the file already imports `System.Collections.Immutable`"** — it does not. The usings are `System`, `System.Collections.Generic`, `System.Text.Json`, `Memex.Portal.Shared.SelfUpdate`, `MeshWeaver.Graph.Configuration`, `Microsoft.Extensions.Configuration`, `Xunit` (`grep -c "System.Collections.Immutable"` on the file → `0`). Taking the suggestion would mean adding an import, not using one.
- **It is not a fixture or held state.** The dictionary is an argument literal consumed by `ConfigurationBuilder.AddInMemoryCollection(IEnumerable<KeyValuePair<string, string?>>)` and unreachable after the `.Build()` on the same expression. The collections policy is about state a type holds — a field, a cache, a repository — where mutation is what bleeds across tests and users; there is nothing here to mutate or to bleed.
- **It is the established idiom, including in this file.** `AddInMemoryCollection(new Dictionary<string, string?> { … })` appears 68 times across `test/`, three of them in this very file (`TheSettings_CarryPresenceOfTheSecret_NeverItsValue`, `TheControlInstance_ReadsAsLocal`, `ABareLocalTarget_IsNotALocalRoute`). Converting only the new one would leave the file inconsistent and assert nothing extra. If the policy is genuinely meant to reach transient builder arguments, that is a fleet-wide ratchet with its own guard, not a one-line divergence smuggled in on a routing fix.

No guard scans for this shape, and the whole project is green (1574 passed).
